src/LegendLanguageClient.ts

```diff
@@ -139,9 +166,243 @@ export class LegendLanguageClient extends LanguageClient {
     token?: CancellationToken,
   ): Promise<string> {
     if (token) {
-      return this.sendRequest(LEGEND_WRITE_ENTITY, request, token);
+      return this.sendRequest(LEGEND_WRITE_ENTITY_REQUEST_ID, request, token);
+    } else {
+      return this.sendRequest(LEGEND_WRITE_ENTITY_REQUEST_ID, request);
+    }
+  }
+
+  async getClassifierPathMap(token?: CancellationToken): Promise<string> {
+    if (token) {
+      return this.sendRequest(GET_CLASSIFIER_PATH_MAP_REQUEST_ID, token);
     } else {
-      return this.sendRequest(LEGEND_WRITE_ENTITY, request);
+      return this.sendRequest(GET_CLASSIFIER_PATH_MAP_REQUEST_ID);
     }
   }
+
+  async getSubtypeInfo(token?: CancellationToken): Promise<string> {
+    if (token) {
+      return this.sendRequest(GET_SUBTYPE_INFO_REQUEST_ID, token);
+    } else {
+      return this.sendRequest(GET_SUBTYPE_INFO_REQUEST_ID);
+    }
+  }
+
+  async analyzeMappingModelCoverage(
+    mappingTextLocation: TextLocation,
+  ): Promise<string> {
+    return commands.executeCommand(
+      LEGEND_COMMAND,
+      mappingTextLocation,
+      ANALYZE_MAPPING_MODEL_COVERAGE_COMMAND_ID,
+    );
+  }
+
+  async executeQuery(
+    entityTextLocation: TextLocation,
+    lambda: V1_RawLambda,
+    mapping: string | undefined,
+    runtime: V1_Runtime | undefined,
+    context: V1_RawExecutionContext,
+    parameterValues: { [key: string]: unknown },
+    serializationFormat?: EXECUTION_SERIALIZATION_FORMAT | undefined,
+  ): Promise<string> {
+    return commands.executeCommand(
+      LEGEND_COMMAND,
+      entityTextLocation,
+      EXECUTE_QUERY_COMMAND_ID,
+      {
+        lambda: JSON.stringify(lambda),
+        mapping,
+        runtime: JSON.stringify(runtime),
+        context: JSON.stringify(context),
+        serializationFormat,
+      },
+      parameterValues,
+    );
+  }
+
+  async exportData(
+    entityTextLocation: TextLocation,
+    lambda: V1_RawLambda,
+    mapping: string | undefined,
+    runtime: V1_Runtime | undefined,
+    context: V1_RawExecutionContext,
+    parameterValues: V1_ParameterValue[],
+    downloadFileName: string,
+    serializationFormat?: EXECUTION_SERIALIZATION_FORMAT | undefined,
+  ): Promise<LegendExecutionResult> {
+    const response = (await commands.executeCommand(
```

Comment:
I'm worry this will lead to OOM.... we likely need to revisit this down the road to push the download / write the file to LSP.